### PR TITLE
discovery: Use getAllSourceChainConfigs

### PR DESCRIPTION
**Pull request by winder**
Allows destination readers to discover all source chain configurations. Previously the algorithm would only discover sources when there was a corresponding contract reader. Now the "getAllSourceChainConfigs"

### CODE REVIEW

File: pkg/reader/ccip_test.go [hunk 1]
```diff
@@ -618,22 +637,24 @@ func TestCCIPChainReader_DiscoverContracts_HappyPath_Round2(t *testing.T) {
...
+	destRouter := []byte{0x6}
 
 	var expectedContractAddresses ContractAddresses
+	for i := range onramps {
+		expectedContractAddresses = expectedContractAddresses.Append(
+			consts.ContractNameOnRamp, sourceChain[i], onramps[i])
+	}
 	// All dest chain contracts should be discovered, they do not require source chain support.
-	//expectedContractAddresses = expectedContractAddresses.Append(consts.ContractNameRouter, destChain, destRouter)
+	expectedContractAddresses = expectedContractAddresses.Append(consts.ContractNameRouter, destChain, destRouter)
```
Comment (winder) [inline pkg/reader/ccip_test.go:653]: Previously dest-only readers were unable to discover their own router :joy:

File: pkg/reader/ccip.go [hunk 2]
```diff
@@ -625,89 +625,88 @@ func (r *ccipChainReader) GetRMNRemoteConfig(
...
 	}
-	resp = resp.Append(consts.ContractNameFeeQuoter, r.destChain, dynamicConfig.FeeQuoter)
 
 	return resp, nil
 }
 
 func (r *ccipChainReader) DiscoverContracts(
 	ctx context.Context,
-	allChains []cciptypes.ChainSelector,
+	_ []cciptypes.ChainSelector, //Deprecated
```
Comment (asoliman92) [inline pkg/reader/ccip.go:700]: Why not remove it in this PR? No external calls for this AFAIK.

File: pkg/reader/ccip.go [hunk 3]
```diff
@@ -972,6 +978,64 @@ func (r *ccipChainReader) getOffRampSourceChainsConfig(
 	return res, nil
 }
 
+// selectorsAndConfigs wraps the return values from getAllSourceChainConfigs.
+type selectorsAndConfigs struct {
+	Selectors          []uint64            `mapstructure:"F0"`
```
Comment (asoliman92) [inline pkg/reader/ccip.go:983]: I believe we don't need to use `mapstructure` here. check this [example](https://github.com/smartcontractkit/chainlink-ccip/blob/ad2992ddf2377fe55c65e1aca9cb03c1c8cd8b2a/internal/reader/home_chain.go#L404-L407) where it's just using normal tag

Comment (winder) [reply pkg/reader/ccip.go:983]: I believe it's actually the json tags in that example which aren't needed. When a tag isn't provided the field name is used.

File: pkg/reader/ccip.go (hunk 2 above)
Comment (winder) [reply pkg/reader/ccip.go]: done

File: pkg/reader/ccip.go [hunk 4]
```diff
@@ -625,89 +625,88 @@ func (r *ccipChainReader) GetRMNRemoteConfig(
...
+		var staticConfig offRampStaticChainConfig
+		err := r.getDestinationData(
+			ctx,
+			chain,
+			consts.ContractNameOffRamp,
+			consts.MethodNameOffRampGetStaticConfig,
+			&staticConfig,
+		)
+		if err != nil {
+			return nil, fmt.Errorf("unable to lookup nonce manager (offramp static config): %w", err)
```
[the remark this answers is not in the document]
Comment (winder) [reply pkg/reader/ccip.go]: done

**Review by asoliman92 — APPROVED** [review]

**Pull request merged**